src/main/scala/com/sky/kafka/configurator/TopicConfigurator.scala

```diff
@@ -9,6 +9,14 @@ import scala.util.{Failure, Success, Try}
 case class TopicConfigurator(topicReader: TopicReader, topicWriter: TopicWriter) extends LazyLogging {
 
   def configure(topic: Topic): Try[Unit] = {
+    tryConfigure(topic).recoverWith {
```

Comment:
i don't really like the name tryConfigure since it has the same signature as configure (they both return `Try[Unit]`).

chaining `recoverWith` like this makes me think that we should handle the logging at a higher level.

I feel like we could change our approach a bit so that the logging is handled in one place (the level above this), will have a think